packages/flutter_tools/test/commands.shard/hermetic/build_ipa_export_plist_test.dart

```diff
@@ -0,0 +1,227 @@
+        buildInfo: BuildInfo.release,
+        bundleId: 'com.example.myapp',
+        codeSignStyle: 'Manual',
+        profileSpecifier: 'MyApp Distribution',
+        teamId: 'ABC123DEF4',
+        profileUuid: '12345678-1234-1234-1234-123456789012',
+        fileSystem: fileSystem,
+      );
+
+      final String plistContent = plistFile.readAsStringSync();
+      expect(plistContent, contains('<string>ad-hoc</string>'));
+      expect(plistContent, contains('<key>teamID</key>'));
+      expect(plistContent, contains('<key>signingStyle</key>'));
+      expect(plistContent, contains('<string>manual</string>'));
+    });
+
+    test('handles null codeSignStyle gracefully', () async {
+      final fileSystem = MemoryFileSystem.test();
+
+      final File plistFile = await BuildIOSArchiveCommand.createExportPlistForTesting(
+        exportMethod: 'app-store',
+        buildInfo: BuildInfo.release,
+        bundleId: 'com.example.myapp',
+        fileSystem: fileSystem,
+      );
+
+      final String plistContent = plistFile.readAsStringSync();
+      // Should generate simple plist when codeSignStyle is null
+      expect(plistContent, contains('<key>method</key>'));
+      expect(plistContent, contains('<string>app-store</string>'));
+      expect(plistContent, isNot(contains('<key>teamID</key>')));
+      expect(plistContent, isNot(contains('<key>signingStyle</key>')));
+    });
+
+    // TODO(mohamed): Integration test for full `flutter build ipa` flow with manual signing.
+    //
+    // This is currently skipped because `IosProject.buildSettingsForBuildInfo` does not
+    // return mocked settings in the test harness environment. The project discovery logic
+    // requires a fully-formed iOS project structure that is difficult to replicate in the
+    // current hermetic test setup.
+    //
+    // **To implement this test, we would need:**
+    // 1. A fully-formed iOS project fixture in the test environment (with proper Xcode
+    //    project structure, pbxproj files, etc.)
+    // 2. Mocking of the `security cms` command for profile decoding
+    // 3. Proper `XcodeProjectInterpreter` mocking that returns build settings
+    // 4. Integration with the full `build ipa` command execution path
+    //
+    // **When implemented, this test should verify:**
+    // - Profile lookup integration with real provisioning profile files
+    // - Full `xcodebuild -exportArchive` flow with generated plist
+    // - Error handling when profile is missing or malformed
+    // - Trace logging output when profile lookup fails
+    // - Successful IPA generation end-to-end
+    //
+    // See https://github.com/flutter/flutter/issues/177853 for the feature issue.
+    // Unit tests above provide sufficient coverage for the plist generation logic itself.
+    test(
+      'full build ipa path generates plist for manual signing (integration)',
+      () {
+        // Integration test skipped - see justification comment below.
```

Comment:
You don't need to leave an empty test in place for the integration test. This comment is better put in the PR description